File: Linear Array/LinearSortedArray.java
public class LinearSortedArray {
    int itemCount;
    int array[];

    public LinearSortedArray(int maxSize){
        array = new int[maxSize];
    }

    public void orderedInsert(int x) {
        if(itemCount < array.length) {
            int insertPos = 0;
            while(insertPos < itemCount && array[insertPos] < x) {
                insertPos++;
            }

            for(int k = itemCount; k > insertPos; k--) {
                array[k] = array[k-1];
            }
            array[insertPos] = x;
            itemCount++;
        }
        else {
            System.out.println("Array is full");
        }
    }

    public void display() {
        for(int i = 0; i < itemCount; i++) {
            System.out.print(array[i] + " ");
        }
    }

    public static void main(String [] args) {
        LinearSortedArray array = new LinearSortedArray(5);
        array.orderedInsert(6);
        array.orderedInsert(5);
        array.orderedInsert(8);
        array.orderedInsert(1);

        array.display();
    }

}

File: Linear Array/LinearSortedArray.cpp
#include <iostream>

using namespace std;

class LinearSortedArray {
public:
	int itemCount;
	int capacity;
	int* myArr;

	LinearSortedArray(int maxSize) {
		capacity = maxSize;
		myArr = new int[maxSize];
	}

	void orderedInsert(int x) {
		if (itemCount < capacity) {
			int insertionPos = 0;
			while (insertionPos < itemCount && myArr[insertionPos] < x) {
				insertionPos++;
			}

			for (int k = itemCount; k > insertionPos; k--) {
				myArr[k] = myArr[k - 1];
			}

			myArr[insertionPos] = x;
			itemCount++;
		}
		else {
			cout << "Array is full" << endl;
		}
		

	}

	void display() {
		for (int i = 0; i < itemCount; i++) {
			cout << myArr[i] << " ";
		}
	}

	~LinearSortedArray() {
		delete myArr;
	}

};

int main() {

	LinearSortedArray arr(5);
	arr.orderedInsert(5);
	arr.orderedInsert(2);
	arr.orderedInsert(9);
	arr.orderedInsert(6);

	arr.display();

	return 0;
}